changebot/github/github_api.py

```diff
@@ -99,6 +110,41 @@ def get_file_contents(self, path_to_file):
         contents_base64 = response.json()['content']
         return base64.b64decode(contents_base64).decode()
 
+    def get_user_config(self, path_to_file='astropybotrules.yaml'):
+        """
+        Load user configuration for bot.
+
+        Parameters
+        ----------
+        path_to_file : str
+            Configuration file in YAML format in the repository.
+            If not given or invalid, default is used.
+
+        Returns
+        -------
+        cfg : dict
+            Configuration parameters.
+
+        """
+        try:
+            file_content = self.get_file_contents(path_to_file)
+            cfg = yaml.load(file_content)
+        except Exception as e:
+            warnings.warn(str(e))
+            # Empty dict means calling code set the default
+            cfg = {}
+
+        return cfg
+
+    def get_config_from_cache(self, cfg_key, cfg_default):
```

Comment:
I would maybe call this ``get_config_value`` - the fact it gets it from cache is an internal optimization.